### ISSUE TITLE
2d text alignment is backwards

### ISSUE THREAD

**Issue by concave-sphere** [bevyengine/bevy#1853]
## Bevy version

0.5

## Operating system & version

Debian GNU/Linux "buster"

## What you did

Inserted text bundles rendering "<HAlign=Left>" with left-alignment and "<HAlign=Right>" with right-alignment, using the same `Transform`.  Did the same for "VAlign\n=\nTop" and "VAlign\n=\nBottom"

## What you expected to happen

The Bevy docs describe left and right alignment as follows:

> `Left`: Leftmost character is immediately to the right of the render position.
> Bounds start from the render position and advance rightwards.
>
> (...)
>
> `Right`: Rightmost character is immediately to the left of the render position.
> Bounds start from the render position and advance leftwards.

They describe top and bottom alignment as follows:

> `Top`: Characters/bounds start underneath the render position and progress downwards.
>
> `Bottom`: Characters/bounds start above the render position and progress upward.

This matches my own expectation based on experience with, e.g, word processors, UI libraries, etc.  As such, I would expect to see something like this in the center of the screen:

```
             VAlign
               =
             Bottom
  <HAlign=Right><HAlign=Left>
             VAlign
               =
              Top
```

Note that the *location* of the text should be the opposite of its alignment.  That is: since the alignment _point_ is in the center of the screen, right-aligned text (text with its right side aligned to the center) should be to the left of center, while left-aligned text (text with its left side aligned to the center) should be to the right of center.

## What actually happened

Layout was exactly the opposite of what I expected:

```
             VAlign
               =
             Top
  <HAlign=Left><HAlign=Right>
             VAlign
               =
              Bottom
```

## Additional information
Program demonstrating the problem: [test.tar.gz](https://github.com/bevyengine/bevy/files/6276083/test.tar.gz)

Screenshot of the test program above on my computer:

![wrong-alignment](https://user-images.githubusercontent.com/64325197/113969955-ce0c8300-97ea-11eb-96e0-97ffe5c5f7cd.png)

**Comment by CaelumLaron** [comment]
Can i try to take this @alice-i-cecile ?

**Comment by alice-i-cecile** [comment]
@CaelumLaron you bet!

**Comment by alice-i-cecile** [comment]
Fixed by #6000.
